### PR TITLE
Initial recipe for tszip.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Re checklist item above, I'm the mentioned maintainer.

File: recipes/tszip/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - pip
+  run:
+    - python
+    - tskit
+    - zarr
+    - humanize
+
+test:
+  commands:
+    - tszip
```
Comment (xhochy) [inline recipes/tszip/meta.yaml]: This fails without a file, call the help to verify it works
```suggestion
    - tszip -h
```

File: recipes/tszip/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - tsunzip
```
Comment (xhochy) [inline recipes/tszip/meta.yaml]:
```suggestion
    - tsunzip -h
```

File: recipes/tszip/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 1edb75b0d8b11c4a7b41ea5d4e14929920b18b3f6a81f3f08c90f0cc80f1d0c1
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  entry_points:
+    - tszip=tszip.cli:tszip_main
+    - tsunzip=tszip.cli:tsunzip_main
+  skip: True  # [py<35]
```
Comment (xhochy) [inline recipes/tszip/meta.yaml]: This is a pure Python package, so making it `noarch` is preferred. The Python version constraint can be expressed below in the `requirements` section
```suggestion
  noarch: Python
```

File: recipes/tszip/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+requirements:
+  host:
+    - setuptools
+    - setuptools_scm
+    - python
```
Comment (xhochy) [inline recipes/tszip/meta.yaml]:
```suggestion
    - python >=3.6
```

File: recipes/tszip/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    - pip
+  run:
+    - python
```
Comment (xhochy) [inline recipes/tszip/meta.yaml]:
```suggestion
    - python >=3.6
```

**Comment by jeromekelleher** [comment]
Thank you so much @xhochy, this is really helpful!

**Pull request merged**